core/physical/include/DiscretizeOperators.h

```diff
@@ -80,7 +82,11 @@ class GPUDownsampleResolution: public PhysicalOperator, public GPUOperator {
             CHECK_LE(theta_samples, std::numeric_limits<unsigned int>::max());
             CHECK_LE(phi_samples, std::numeric_limits<unsigned int>::max());
 
-            LOG(INFO) << "Downsampling to " << theta_samples << 'x' << phi_samples;
+            if (theta_samples != last_downsampled_theta_ || phi_samples != last_downsampled_phi_)
```

Comment:
Can we construct a `Configuration` object and then compare it to the memoized version?

Reply:
I am making it so that we memoize the downsampled configuration and remember the latest base configuration so this computation & logging is skipped when the base configuration doesn't change. If the base configuration changes, I think it's ok to print the logging statement again because it seems like that will be a rare occurrence.

Reply:
Right, totally agree on the logging. I'm suggesting something like:

```
Configuration proposed_configuration = ...
if(proposed_configuration != previous_configuration)
    LOG(...) ...
```

Reply:
I'll add that check 👍

Reply:
I think there's a convenience constructor for creating a `Configuration` instance from a template with a new height/width.

Reply:
It looks like `EncodeConfiguration` has that constructor, but I can move it into `Configuration` if you think that would be better than specifying all of the fields.

Reply:
Nah best to leave `Configuration` as a simple struct.